src/std/io/strio/api.ss

```diff
@@ -0,0 +1,253 @@
+;;; -*- Gerbil -*-
+;;; © vyzo
+;;; string IO api
+(import :std/interface
+        ../interface
+        ../dummy
+        ../bio/types
+        ./types
+        ./input
+        ./output
+        ./inline
+        ./reader
+        ./utf8
+        ./writer
+        ./chunked
+        ./delimited
+        ./util)
+(export open-string-reader
+        open-string-writer
+        open-buffered-string-reader
+        open-buffered-string-writer
+        get-buffer-output-string
+        get-buffer-output-string-chunks
+        (import: ./inline)
+        (import: ./util))
+
+(def default-u8vector-buffer-size (expt 2 15)) ; 32KB
+(def default-string-buffer-size (expt 2 13)) ; 8k chars
+(def max-u8vector-buffer-size (expt 2 28)) ; 256MB -- for safe efficient packing in 64bit archs
+
+(def (make-u8vector-buffer buffer-or-size)
+  (cond
+   ((fixnum? buffer-or-size)
+    (if (fx> buffer-or-size max-u8vector-buffer-size)
+      (error "Bad argument; buffer too big" buffer-or-size)
+      (make-u8vector buffer-or-size)))
+   ((u8vector? buffer-or-size)
+    (if (fx> (u8vector-length buffer-or-size) max-u8vector-buffer-size)
+      (error "Bad argument; buffer too big" (u8vector-length buffer-or-size))
+      buffer-or-size))
+   (else
+    (error "Bad argument; expected fixnum or u8vector" buffer-or-size))))
+
+(def (make-string-buffer buffer-or-size)
+  (cond
+   ((fixnum? buffer-or-size)
+    (if (fx> buffer-or-size max-u8vector-buffer-size)
+      (error "Bad argument; buffer too big" buffer-or-size)
+      (make-string buffer-or-size)))
+   ((string? buffer-or-size)
+    (if (fx> (string-length buffer-or-size) max-u8vector-buffer-size)
+      (error "Bad argument; buffer too big" (string-length buffer-or-size))
+      buffer-or-size))
+   (else
+    (error "Bad argument; expected fixnum or string" buffer-or-size))))
+
+(def (double string-buffer-or-size)
+  (fx* 2 (if (fixnum? string-buffer-or-size)
+           string-buffer-or-size
+           (string-length string-buffer-or-size))))
+
+(def (character-encoder codec)
+  (case codec
+    ((UTF-8) utf8-encode-partial!)
+    (else
+     (error "Unsupported character encoding" codec))))
+
+(def (character-decoder codec)
+  (case codec
+    ((UTF-8) utf8-decode-partial!)
+    (else
+     (error "Unsupported character encoding" codec))))
+
+(def (open-string-reader reader (buffer-or-size default-u8vector-buffer-size)
+                         encoding: (codec 'UTF-8))
+  (cond
+   ((is-BufferedReader? reader)
```

Comment:
It is different from `BufferedReader?`; that tests for exact interface instance type.

The `is-` predicates check for castability, so they do a different thing. We could perhaps name them `implements-X?` as it would better convey intent. Feel free to do a follow up pr doing that if you feel strongly about it.